lib/tree.rb

```diff
@@ -19,44 +19,119 @@ def initialize
   # Time Complexity: 
   # Space Complexity: 
   def add(key, value)
-    raise NotImplementedError
+    new_node = TreeNode.new(key, value)
+    @root.nil? ? @root = new_node : comparative_helper(@root, new_node)
+  end
+
+  def comparative_helper(current_node, new_node)
+    return new_node if current_node.nil?
+
+    if new_node.key <= current_node.key
+      current_node.left = comparative_helper(current_node.left, new_node)
+    else
+      current_node.right = comparative_helper(current_node.right, new_node)
+    end
+
+    return current_node
   end
 
   # Time Complexity: 
   # Space Complexity: 
   def find(key)
-    raise NotImplementedError
+    return nil if @root.nil?
+    return @root.value if @root.key == key
+    
+    current_node = @root
+
+    until current_node == nil 
+      return current_node.value if key == current_node.key
+      current_node = current_node.left if key < current_node.key
+      current_node = current_node.right if key > current_node.key
+    end 
+
   end
 
   # Time Complexity: 
   # Space Complexity: 
   def inorder
```

Comment:
👍 , But what about time/space complexity.

Wow, doing a loop solution for inorder.